CDCgov/prime-reportstream
# 6103 FHIR to HL7 library converts repeating segments

This PR allows the FHIR to HL7 library to convert FHIR data to repeating HL7 segments.

Test Steps:
1. Check out the branch for this PR.
2. Verify the tests pass by running the command `./gradlew package` and verifying the build is successful.
3. Run the following command to convert a FHIR bundle to HL7.  Note the conversion at this point is minimal, but seeing that an HL7 message is generated is the point.  
`./prime fhirdata --input-file "src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir" -s "src/test/resources/fhirengine/translation/hl7/schema/schema-read-test-01/ORU_R01.yml"`
4. Verify an HL7 message is generated from the conversion and includes repeating segments like ORC, OBR, OBX.  You will see an output like this:

## Changes
- Added code to handle repeating HL7 segments using the resourceIndex property
- Added the use of constants

## Checklist

### Testing

### Process

## Linked Issues
- #6103

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
